File: 226. Invert Binary Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def invertTree(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        p = root
        self.helper(p)
        return root
        
    def helper(self, root):
        if not root:
            return []
        temp = root.left
        root.left = root.right
        root.right = temp
        self.helper(root.left)
        self.helper(root.right)

File: 226. Invert Binary Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {
    var p = root;
    var helper = function(p) {
        if (p === null)
            return [];
        var temp = p.left;
        p.left = p.right;
        p.right = temp;
        helper(p.left);
        helper(p.right);
    };
    helper(p);
    return root;
};
